main.cpp

```diff
@@ -367,6 +342,27 @@ FloatType& FloatType::pow(const IntType& it)
     return powInternal(static_cast<float>(it));
 }
 
+FloatType& FloatType::apply(std::function<FloatType&(float& f)> func)
+{
+    if(func)
+        func(*value);
```

Comment:
the lambda returns a reference to `FloatType&`. 
This function also returns a reference to `FloatType&`. 
```
 2) One of the apply() functions should takes a std::function<> object as the function argument.
    the std::function<> object should return *this;
```
if `func` is valid, then return the result that `func` returns from this function. 
- [ ] revise